### PR TITLE
Picker haptics

### PR DESCRIPTION
**Pull request by Kpeved**
#### WHAT
Haptics threshold was exposed in rememberRotaryHapticHandler api. 
Haptics were also added into pickerGroupItemWithRSB for better usability

#### WHY
There were no haptics in pickerGroupItemWithRSB

#### HOW
rememberRotaryHapticHandler was used for triggering haptics asynchronously 

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt [hunk 1]
```diff
@@ -206,17 +206,25 @@ public fun rememberDisabledHaptic(): RotaryHapticHandler = remember {
...
+ * @param hapticsThreshold A scroll threshold after which haptic is produced.
+ * @param hapticsChannel Channel to which haptic events will be sent
+ * @param rotaryHaptics Interface for Rotary haptic feedback which performs haptics
  */
 @ExperimentalHorologistApi
 @Composable
 public fun rememberRotaryHapticHandler(
     scrollableState: ScrollableState,
     throttleThresholdMs: Long = 40,
+    hapticsThreshold: Long = 50,
```
Comment (yschimke) [inline compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt:222]: What about `hapticsThresholdPx`

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt [hunk 2]
```diff
@@ -206,17 +206,25 @@ public fun rememberDisabledHaptic(): RotaryHapticHandler = remember {
 
 /**
  * Remember rotary haptic handler.
+ * @param scrollableState A scrollableState, used to determine whether the end of the scrollable
+ * was reached or not.
+ * @param throttleThresholdMs Throttling events within specified timeframe.
+ * Only first and last events will be received. Check [throttleLatest] for more info.
+ * @param hapticsThreshold A scroll threshold after which haptic is produced.
```
Comment (yschimke) [inline compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt:213]: it's unclear whether this is dp, px or ms.

File: composables/src/main/java/com/google/android/horologist/composables/TimePicker.kt [hunk 3]
```diff
@@ -525,13 +526,19 @@ internal fun pickerGroupItemWithRSB(
     option: @Composable PickerScope.(optionIndex: Int, pickerSelected: Boolean) -> Unit
 ): PickerGroupItem {
     val coroutineScope = rememberCoroutineScope()
+    val haptics = rememberRotaryHapticHandler(
+        scrollableState = pickerState,
+        throttleThresholdMs = 10
+    )
```
Comment (yschimke) [inline composables/src/main/java/com/google/android/horologist/composables/TimePicker.kt:532]: Is the default px threshold of 50 still right here?

**Review by yschimke — APPROVED** [review]
Looks great!

**Comment by yschimke** [comment]
@rajat4064g should approve, consider mine a +1, unless you need to land urgently.

**Review by rajat4064g — APPROVED** [review]
LGTM

**Pull request merged**